examples/cndpfwd/l3-fwd.c

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 Intel Corporation.
+ */
+
+#include <cne_fib.h>           // For fib_create, fib_add...
+#include <cne_inet4.h>         // for inet_mtoh64 and inet_h64tom
+#include <net/cne_ether.h>     // for ether_addr
+
+#include "main.h"
+
+#define IPV4_L3FWD_NUM_FIB_RULES \
+    (sizeof(ipv4_l3fwd_fib_rule_array) / sizeof(ipv4_l3fwd_fib_rule_array[0]))
+
+struct ipv4_l3fwd_fib_rule {
+    uint32_t ip;
+    uint8_t depth;
+    struct ether_addr nh;
+    uint16_t tx_port;
+};
+
+static struct cne_fib *fib;
+
+static int
+l3fwd_fib_populate(struct cne_fib *fib)
+{
+    /*
+     * 198.18.0.0/16 are set aside for RFC2544 benchmarking (RFC5735).
+     * Each FIB rule is IPv4, depth, next hop destination MAC, TX port.
+     */
+    struct ipv4_l3fwd_fib_rule ipv4_l3fwd_fib_rule_array[] = {
```

Comment:
Not a big fan of having these constant values in the source code. One option is to add a new key value pair array to the .jsonc file in the **options** section. This was done in the examples/cnet-graph jsonc file for adding socket connections.

I was thinking we can add a new array in the options section for this rules.

// fib-rules define the IP address, MAC address and port index for the FIB table entries
//    Three parts to each entry in the array. IP address with mask value/depth, MAC address and port index.
"fib-rules": [
    "198.18.0.0/24,02:00:01:02:03:04,1",
    "198.18.1.0/24,06:00:01:02:03:04,0",
]

The above is just an example and we can format the strings as you see fit. The example/cnet-graph has an example of how to access the options:graph:0-chanl strings.

Then we can remove hardcoded values from this file.

Reply:
We did this because it was a little easier to implement (and after discussion with @jeffreybshaw who pointed me to similar behavior in https://github.com/DPDK/dpdk/blob/main/examples/l3fwd/main.c#L204).

Totally agree that parsing the rules from config would be cleaner, though. Want us to do that on this PR, or open an enhancement issue once this one is merged?

Reply:
Have a look and see if you think opening a new PR or enhancement is reasonable, I would prefer it fixed in this PR. I can understand if you want to leave it this way and create a new PR or enhancement. If you are going to create a new PR please create an enhancement for the change.